ingestion_server/ingestion_server/cleanup.py

```diff
@@ -272,24 +272,72 @@ def _clean_data_worker(rows, temp_table, sources_config, all_fields: list[str]):
     return cleaned_values
 
 
-def save_cleaned_data(result: dict) -> dict[str, int]:
-    log.info("Saving cleaned data...")
-    start_time = time.time()
+class CleanDataUploader:
+    # Number of records to buffer in memory before writing to disk
+    mem_buffer_size = 100
 
-    cleanup_counts = {field: len(items) for field, items in result.items()}
-    for field, cleaned_items in result.items():
-        # Skip the tag field because the file is too large and fills up the disk
-        if field == "tag":
-            continue
-        if cleaned_items:
-            with open(f"{field}.tsv", "a") as f:
-                csv_writer = csv.writer(f, delimiter="\t")
-                csv_writer.writerows(cleaned_items)
+    # Size (in MB) of local file buffer before writing to S3
+    disk_buffer_size = 10 * 1024 * 1024
 
-    end_time = time.time()
-    total_time = end_time - start_time
-    log.info(f"Finished saving cleaned data in {total_time},\n{cleanup_counts}")
-    return cleanup_counts
+    buffer = {
+        field: {"part": 1, "rows": []}
+        for field in _cleanup_config["tables"]["image"]["sources"]["*"]["fields"]
+    }
+
+    def __init__(self):
+        bucket_name = config("AWS_S3_BUCKET", default="openverse-catalog")
+        self.s3 = self._get_s3_resource()
+        self.s3_bucket = self.s3.Bucket(bucket_name)
+
+    @staticmethod
+    def _get_s3_resource():
+        if config("ENVIRONMENT") == "local":
+            return boto3.resource(
+                "s3",
+                endpoint_url=config("AWS_S3_ENDPOINT", default="http://s3:5000"),
+                aws_access_key_id=config("AWS_ACCESS_KEY_ID", default="test_key"),
+                aws_secret_access_key=config(
+                    "AWS_SECRET_ACCESS_KEY", default="test_secret"
+                ),
+            )
+
+        return boto3.resource(
+            "s3", region_name=config("AWS_REGION", default="us-east-1")
+        )
+
+    def _save_to_disk(self, field: str, force_upload=False):
+        log.info(f"Saving {len(self.buffer[field])} rows of `{field}` to local file.")
+        with open(f"{field}.tsv", "a") as f:
+            csv_writer = csv.writer(f, delimiter="\t")
+            csv_writer.writerows(self.buffer[field]["rows"])
+            # Clean memory buffer of saved rows
+            self.buffer[field]["rows"] = []
+
+        if os.path.getsize(f"{field}.tsv") >= self.disk_buffer_size or force_upload:
```

Comment:
Right about the variable, added ✔️ 

I think the directory is indifferent since it has been working well this way, but if you prefer a specific location, we can discuss it.